### PR TITLE
Fix Dart 2 issues in firebase plugins

### PR DESCRIPTION
**Pull request by mraleph**
Fixes https://github.com/flutter/flutter/issues/15027

### CODE REVIEW

**Comment by mraleph** [comment]
/fyi @mravn-google I have fixed all `firebase_*` plugins except `firebase_database` where test crashes and I don't have time for it right now.

**Comment by tvolkert** [comment]
Looks like Travis isn't happy:

```
Checking project firebase_auth...
  error - Missing parameter type for 'v' at lib/firebase_auth.dart:111:71 - (strong_mode_implicit_dynamic_parameter)
  info - Specify type annotations at lib/firebase_auth.dart:111:71 - (always_specify_types)
  info - Avoid using `as` at lib/firebase_auth.dart:111:77 - (avoid_as)
```

File: packages/firebase_auth/lib/firebase_auth.dart [hunk 1]
```diff
@@ -108,7 +108,11 @@ class FirebaseAuth {
 
     StreamController<FirebaseUser> controller;
     controller = new StreamController<FirebaseUser>.broadcast(onListen: () {
-      _handle = channel.invokeMethod('startListeningAuthState').then((v) => v as int);
+      _handle =
```
Comment (mravn-google) [inline packages/firebase_auth/lib/firebase_auth.dart]: Could we do
```dart
_handle = channel.invokeMethod('startListeningAuthState').then<int>((dynamic v) => v);
```

Comment (mraleph) [reply packages/firebase_auth/lib/firebase_auth.dart]: Done.

**Comment by mraleph** [comment]
I have fixed more issues. Where I fixed the plugin code itself I bumped version in pubspec and added an entry to CHANGELOG. 

Now all the tests except one test in firebase_database pass. The test in firebase_database fails due to https://github.com/dart-lang/sdk/issues/28498. 

@mravn-google please take another quick look

**Comment by mraleph** [comment]
@tvolkert Travis seems to be stuck on one of the Shards - do you know if there is a way to nudge it?

**Comment by mravn-google** [comment]
I restarted the job.

**Comment by mraleph** [comment]
Now, I am getting errors in the files I did not edit :-/ I am just gonna push it as is.

**Comment by tvolkert** [comment]
It looks like `dartfmt` has been updated with some improvements, but it's yielding an error because our existing formatted code isn't formatted correctly anymore.

**Comment by cdock1029** [comment]
When will this be available through pub? Not seeing 0.4.7 yet

https://pub.dartlang.org/packages/firebase_auth#-versions-tab-

**Comment by mravn-google** [comment]
Sorry, that was an omission on my part. Version 0.4.7 has just been published.

**Pull request merged**